takari-lifecycle-plugin/src/test/java/io/takari/maven/plugins/compile/CompileTest.java

```diff
@@ -365,4 +365,35 @@ public void testInnerTypeDependency_sourceDependencies() throws Exception {
     mojos.compile(project, newParameter("dependencySourceTypes", "prefer"));
     mojos.assertBuildOutputs(new File(basedir, "target/classes"), "innertyperef/InnerTypeRef.class");
   }
+
+  @Test
+  public void testFailOnErrorTrue() throws Exception {
+    try{
+      compile("compile/failOnError");
+      Assert.fail();
+    } catch (MojoExecutionException e) {
+      if(compilerId.equals("jdt")){
+        Assert.assertTrue(e.getMessage().contains("Error.java:[5:13] Strin cannot be resolved to a type"));
```

Comment:
don't assert exception message, assert build context message. There is also helper to deal with different messages between javac and jdt.